features/dd-sdk-android-trace/src/main/java/com/datadog/opentracing/propagation/DatadogHttpCodec.java

```diff
@@ -77,13 +78,11 @@ public void inject(final DDSpanContext context, final TextMapInject carrier) {
             tags.append(MOST_SIGNIFICANT_TRACE_ID_TAG_KEY);
             tags.append('=');
             tags.append(mostSignificantTraceId);
+            carrier.put(DATADOG_TAGS_KEY, tags.toString());
+
             if(sessionId != null) {
-                tags.append(',');
-                tags.append(RUM_SESSION_ID_TAG_KEY);
-                tags.append('=');
-                tags.append(sessionId);
+                carrier.put(BAGGAGE_KEY, RUM_SESSION_ID_BAGGAGE_KEY + '='+sessionId);
```

Comment:
do we need to have a similar change for the `com.datadog.trace.core.propagation.DatadogHttpCodec`, etc.?

Reply:
Indeed, done